### DISCUSSION TITLE
Sentry routing instrumentation?

### DISCUSSION THREAD

**Discussion by RIP21** [TanStack/router#714] [Q&A]
Integration with Sentry routing instrumentation?

React router has a default integration with Sentry, wondering if anything is needed for tanstack router, or if it will just work out of the box, or if not, any hints on which APIs I can use from tanstack router to implement it myself for the now.

Here is how it looks for react-router
https://docs.sentry.io/platforms/javascript/guides/react/configuration/integrations/react-router/

**Comment by maxstue** [comment]
@RIP21 Hey did you find a solution or anything else that would help with implementing this myself ?

**Reply by RIP21** [reply]
Unfortunately my app isn't used at all to get any usage statistics so I can get you any reasonable answer :) So, no, no solution.

**Reply by maxstue** [reply]
ouh okay, but have you tried something that is useable?

**Reply by RIP21** [reply]
No

**Comment by IanVS** [comment]
Sentry recommends creating your own integration based on the react-router one they have: https://github.com/getsentry/sentry-javascript/issues/7927#issuecomment-1936379075

**Comment by lforst** [comment]
For anybody subscribed to this discussion, Sentry now has first-class TanStack Router support! https://docs.sentry.io/platforms/javascript/guides/react/features/tanstack-router/
